bcda/auth/ssas.go

```diff
@@ -14,9 +16,31 @@ type SSASPlugin struct {
 }
 
 // RegisterSystem adds a software client for the ACO identified by localID.
-func (s SSASPlugin) RegisterSystem(localID string) (Credentials, error) {
-	// s.client.CreateSystem()
-	return Credentials{}, errors.New("not yet implemented")
+func (s SSASPlugin) RegisterSystem(localID, publicKey string) (Credentials, error) {
+	creds := Credentials{}
+	aco, err := GetACOByClientID(localID)
+	if err != nil {
+		return creds, errors.Wrap(err, "failed to create system")
+	}
+	trackingID := uuid.NewRandom().String()
+
+	cb, err := s.client.CreateSystem(
+		aco.Name,
+		*aco.CMSID,
+		os.Getenv("SSAS_DEFAULT_SYSTEM_SCOPE"),
+		publicKey,
+		trackingID,
+	)
+	if err != nil {
+		return creds, errors.Wrap(err, "failed to create system")
+	}
+
+	err = json.Unmarshal(cb, &creds)
+	if err != nil {
+		return creds, errors.Wrap(err, "failed to create system")
```

Comment:
`errors.Wrap(err, "msg")` should still wrap the original specific error with a stack trace and the additional supplied message so we should be able to see the specific error and where it comes from.  see: https://godoc.org/github.com/pkg/errors#hdr-Adding_context_to_an_error

Do you still think we should put something else in the message?

Reply:
`errors.Wrap()` records the stack trace, but the caller getting the error back would have to specifically print the extended value (`%+v`) for it to appear, which we don't really do anywhere (yet?)

Reply:
Error messages have a purpose; they should help whoever receives them to understand why the error occurred, how serious the error is,  what kind of action they should take in response, and whether they should add preventive measure and additional tests to the code. In this case, since a system is actually created, the error message is incorrect and misleading. It could easily lead the reader to an erroneous conclusion that it was a benign failure, since nothing was created. The natural response of someone receiving it will likely be to retry the op, in which case we will have two otherwise identical systems. When we add checks for expired systems, the two systems will show up, causing confusion. If code is written in BCDA that assumes only one system exists for an ACO --- in other words, the result of a GET on /groups returns a list containing one system --- what's going to happen when they get two? I'm not going to go on about potential consequences.